x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -375,25 +374,31 @@ export function App({
 
     const originalInput = saveProps.newCopyOnSave ? undefined : initialInput;
     const originalSavedObjectId = (originalInput as LensByReferenceInput)?.savedObjectId;
-    if (options.saveToLibrary && !originalInput) {
```

Comment:
Removed `&& !originalInput` because this actually caused a title naming conflict without warning users.

Steps to view:
1) create and save a lens saved object called "alpha"
2) create and save another lens saved object called "bravo"
3) edit the first lens saved object. Click save and change the title to "bravo". Click save. The modal should notify user about duplicate title but it does not.